src/client/array/dc_array.c

```diff
@@ -175,6 +179,10 @@ free_io_params_cb(tse_task_t *task, void *data)
 	while (io_list) {
 		struct io_params *current = io_list;
 
+		if (current->iom.iom_recxs) {
+			D_FREE(current->iom.iom_recxs);
+			current->iom.iom_recxs = NULL;
```

Comment:
D_FREE sets the pointer to NULL

Reply:
OK, will wait for more comments from others before refreshing.